### PR TITLE
Wii port

### PR DESCRIPTION
**Pull request by askotx**
Is this ok to merge? Or should I make a PR for each one?

This is how it looks on 4:3 TV-> Black border is a little bigger, it looks like GBA Player
sam_1005

### CODE REVIEW

File: src/platform/wii/main.c [hunk 1]
```diff
@@ -291,7 +296,9 @@ void _guiPrepare(void) {
 
 void _guiFinish(void) {
 	Mtx44 proj;
-	guOrtho(proj, -10, VIDEO_VERTICAL_PIXELS + 10, 0, VIDEO_HORIZONTAL_PIXELS, 0, 300);
+	short top = (CONF_GetAspectRatio() == CONF_ASPECT_16_9) ? 10 : 20;
+	short bottom = VIDEO_VERTICAL_PIXELS + top;
+	guOrtho(proj, -(top), bottom, 0, VIDEO_HORIZONTAL_PIXELS, 0, 300);
```
Comment (endrift) [inline src/platform/wii/main.c]: No need for parens around `-top`.

File: src/platform/wii/main.c [hunk 2]
```diff
@@ -42,6 +42,9 @@ static void _gameUnloaded(struct GBAGUIRunner* runner);
 static void _drawFrame(struct GBAGUIRunner* runner, bool faded);
 static uint16_t _pollGameInput(struct GBAGUIRunner* runner);
 
+static s8 WPAD_StickX(u8 chan,u8 right);
+static s8 WPAD_StickY(u8 chan,u8 right);
```
Comment (endrift) [inline src/platform/wii/main.c]: Where do these functions get defined?

**Comment by endrift** [comment]
Really, it should be two PRs, but the individual commits are small enough that it probably won't matter.

File: src/platform/wii/main.c (hunk 2 above)
Comment (askotx) [inline src/platform/wii/main.c]: Damn  ... you're right, I forget to add them ... I'll add them.

**Comment by askotx** [comment]
Sorry for my mistakes.

**Comment by endrift** [comment]
This PR is fine. It gets updated whenever you push to the branch the PR was made on. Last remaining bits to fix are style bits.

File: src/platform/wii/main.c [hunk 3]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
 	UNUSED(source);
 	return gyroZ;
 }
+
+static s8 WPAD_StickX(u8 chan,u8 right){
```
Comment (endrift) [inline src/platform/wii/main.c]: Space after comma, space before brace.

File: src/platform/wii/main.c [hunk 4]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	float mag = 0.0;
+	float ang = 0.0;
+	WPADData *data = WPAD_Data(chan);
+
+	switch (data->exp.type)	{
+
```
Comment (endrift) [inline src/platform/wii/main.c]: No extra line here.

File: src/platform/wii/main.c [hunk 5]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+		case WPAD_EXP_NUNCHUK:
+		case WPAD_EXP_GUITARHERO3:
+			if (right == 0){
```
Comment (endrift) [inline src/platform/wii/main.c]: Space after brace.

File: src/platform/wii/main.c [hunk 6]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+				mag = data->exp.nunchuk.js.mag;
+				ang = data->exp.nunchuk.js.ang;
+			}
+			break;
+
+		case WPAD_EXP_CLASSIC:
+			if (right == 0){
```
Comment (endrift) [inline src/platform/wii/main.c]: Space after brace.

File: src/platform/wii/main.c [hunk 7]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+				mag = data->exp.classic.ljs.mag;
+				ang = data->exp.classic.ljs.ang;
+			}
+			else{
```
Comment (endrift) [inline src/platform/wii/main.c]: Space after brace, and closing brace on same line as else.

File: src/platform/wii/main.c [hunk 8]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
 }
+
+static s8 WPAD_StickX(u8 chan,u8 right){
+	float mag = 0.0;
+	float ang = 0.0;
+	WPADData *data = WPAD_Data(chan);
+
+	switch (data->exp.type)	{
+
+		case WPAD_EXP_NUNCHUK:
```
Comment (endrift) [inline src/platform/wii/main.c]: Do not indent an additional level for the contents of the switch statement.

File: src/platform/wii/main.c [hunk 9]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+				ang = data->exp.classic.rjs.ang;
+			}
+			break;
+
+		default:
+			break;
+	}
+
+	/* calculate X value (angle need to be converted into radian) */
+	if (mag > 1.0) mag = 1.0;
```
Comment (endrift) [inline src/platform/wii/main.c]: if statements needs braces and a line for each statement.

File: src/platform/wii/main.c [hunk 10]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	else if (mag < -1.0) mag = -1.0;
```
Comment (endrift) [inline src/platform/wii/main.c]: Ditto.

File: src/platform/wii/main.c [hunk 11]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	double val = mag * sin(M_PI * ang/180.0f);
```
Comment (endrift) [inline src/platform/wii/main.c]: Should be `sinf(M_PI * ang / 180.0f)`

File: src/platform/wii/main.c [hunk 12]
```diff
@@ -504,3 +513,81 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+ 
+	return (s8)(val * 128.0f);
+}
+
+
+static s8 WPAD_StickY(u8 chan, u8 right)
```
Comment (endrift) [inline src/platform/wii/main.c]: Each of the style nitpicks above are applicable here too.

Comment (askotx) [inline src/platform/wii/main.c]: °_°7

File: src/platform/wii/main.c [hunk 13]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
 	UNUSED(source);
 	return gyroZ;
 }
+
+static s8 WPAD_StickX(u8 chan, u8 right) {
+
```
Comment (endrift) [inline src/platform/wii/main.c]: Remove extra line.

File: src/platform/wii/main.c [hunk 14]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	switch (data->exp.type)	{
+
+	case WPAD_EXP_NUNCHUK:
+	case WPAD_EXP_GUITARHERO3:
+		if (right == 0) {
+			mag = data->exp.nunchuk.js.mag;
+			ang = data->exp.nunchuk.js.ang;
+		}
+	break;
+
```
Comment (endrift) [inline src/platform/wii/main.c]: Remove extra line.

File: src/platform/wii/main.c [hunk 15]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	case WPAD_EXP_CLASSIC:
+		if (right == 0) {
+			mag = data->exp.classic.ljs.mag;
+			ang = data->exp.classic.ljs.ang;
+		} else {
+			mag = data->exp.classic.rjs.mag;
+			ang = data->exp.classic.rjs.ang;
+		}
+	break;
+
```
Comment (endrift) [inline src/platform/wii/main.c]: Remove extra line and indent break.

File: src/platform/wii/main.c [hunk 16]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+	default:
+	break;
```
Comment (endrift) [inline src/platform/wii/main.c]: Indent break.

File: src/platform/wii/main.c [hunk 17]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+		mag = -1.0;
+	}
+	double val = mag * sinf(M_PI * ang / 180.0f);
+ 
+	return (s8)(val * 128.0f);
+}
+
+
+static s8 WPAD_StickY(u8 chan, u8 right) {
+
```
Comment (endrift) [inline src/platform/wii/main.c]: Same nits as above.

File: src/platform/wii/main.c [hunk 18]
```diff
@@ -504,3 +513,86 @@ int32_t _readGyroZ(struct GBARotationSource* source) {
...
+			ang = data->exp.classic.rjs.ang;
+		}
+	break;
+
+	default:
+	break;
+	}
+
+	/* calculate X value (angle need to be converted into radian) */
+	if (mag > 1.0){
```
Comment (endrift) [inline src/platform/wii/main.c]: Add missing space between `)` and `{`

**Comment by endrift** [comment]
Looks like this is almost ready to merge, only a handful of style nits left.

**Comment by askotx** [comment]
OK, I'll do it!

**Comment by endrift** [comment]
You've still got the extra lines in the `switch` statements, but after you fix those and squash, this'll be ready to merge. Thanks!

**Comment by endrift** [comment]
Great, thanks!

**Pull request merged**